instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/event/CustomSpanEventTest.java

```diff
@@ -0,0 +1,121 @@
+                                  // span when using spring-batch versions [3, 4)
+                                  // that bug was fixed in 4.0
+                                  if (VERSION_GREATER_THAN_4_0) {
+                                    assertThat(spanData)
+                                        .hasEventsSatisfyingExactly(
+                                            event -> event.hasName("step.before"),
+                                            event -> event.hasName("step.after"));
+                                  } else {
+                                    assertThat(spanData)
+                                        .hasEventsSatisfyingExactly(
+                                            event -> event.hasName("step.before"),
+                                            event -> event.hasName("chunk.before"),
+                                            event -> event.hasName("chunk.after"),
+                                            event -> event.hasName("step.after"));
+                                  }
+                                }),
+                    span ->
+                        span.hasName(
+                                "BatchJob customSpanEventsItemsJob.customSpanEventsItemStep.Chunk")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasParent(trace.getSpan(1))
+                            .satisfies(
+                                spanData -> {
+                                  // CompositeChunkListener has broken ordering that causes
+                                  // listeners that do not override order() to appear first at all
+                                  // times
+                                  // because of that a custom ChunkListener will always see a Step
+                                  // span when using spring-batch versions [3, 4)
+                                  // that bug was fixed in 4.0
+                                  if (VERSION_GREATER_THAN_4_0) {
+                                    assertThat(spanData)
+                                        .hasEventsSatisfyingExactly(
+                                            event -> event.hasName("chunk.before"),
+                                            event -> event.hasName("chunk.after"));
+                                  } else {
+                                    assertThat(spanData.getEvents()).isEmpty();
+                                  }
+                                }),
+                    span -> {}, // ignore
+                    span -> {}, // ignore
+                    span -> {}, // ignore
+                    span -> {} // ignore
+                    )));
+  }
+
+  protected void itemSpans(TraceAssert trace) {
+    trace.hasSpansSatisfyingExactly(
+        span -> {}, // already checked in the outer method
+        span -> {}, // already checked in the outer method
+        span -> {}, // already checked in the outer method
```

Comment:
usually we use something like
```java
List<Consumer<SpanDataAssert>> assertions = new ArrayList<>();
assertions.addAll(Arrays.asList(span -> ...));
itemSpans(assertions);
trace.hasSpansSatisfyingExactly(assertions);
```
when we need conditionally add assertions or split them for some other reason